reverse.jl

```diff
@@ -0,0 +1,68 @@
+function EnzymeRules.augmented_primal(
+    config,
+    func::Const{typeof(cholesky)},
+    RT::Type{<:Union{Const, DuplicatedNoNeed, Duplicated}},
+    A::Union{Const, Duplicated};
+    kwargs...
+)
+    fact = cholesky(A.val; kwargs...)
+    dA = similar(fact.factors)
+    println("Custom Cholesky augmented forward rule")
+    # dfact would be a dense matrix
+    if needs_primal(config)
+        return AugmentedReturn(fact, dfact, (fact, dfact, dA))
+    else
+        return AugmentedReturn(nothing, (bx, bstats), (fact,))
+    end
+end
+
+function EnzymeRules.augmented_primal(
```

Comment:
We already have a rule for \, see here: https://github.com/EnzymeAD/Enzyme.jl/blob/8fb0f1000380187f395258d08b727a15653cf493/src/internal_rules.jl#L311C59-L311C68

Reply:
That is for a matrix as an input. That's slow if you call it multiple times as you refactorize. This should take the factorization (and work with other factorizations).